openmls/src/credentials/mod.rs

```diff
@@ -197,47 +244,22 @@ impl Credential {
     }
 
     /// Creates and returns a new [`Credential`] of the given
-    /// [`CredentialType`] for the given identity.
-    /// If the credential holds key material, this is generated and stored in
-    /// the key store.
+    /// the serialized credential.
     ///
     /// Returns an error if the given [`CredentialType`] is not supported.
-    pub fn new(
-        identity: Vec<u8>,
-        credential_type: CredentialType,
-    ) -> Result<Self, CredentialError> {
-        let mls_credential = match credential_type {
-            CredentialType::Basic => BasicCredential {
-                identity: identity.into(),
-            },
-            _ => return Err(CredentialError::UnsupportedCredentialType),
-        };
-        let credential = Credential {
-            credential_type,
-            credential: MlsCredentialType::Basic(mls_credential),
-        };
-        Ok(credential)
-    }
-
-    /// Returns the identity of a given credential.
-    pub fn identity(&self) -> &[u8] {
-        match &self.credential {
-            MlsCredentialType::Basic(basic_credential) => basic_credential.identity.as_slice(),
-            // TODO: implement getter for identity for X509 certificates. See issue #134.
-            MlsCredentialType::X509(_) => panic!("X509 certificates are not yet implemented."),
+    pub fn new(serialized_credential: Vec<u8>) -> Self {
+        Self {
+            serialized_credential_content: serialized_credential.into(),
+            credential_type: CredentialType::Basic,
         }
     }
-}
 
-impl From<MlsCredentialType> for Credential {
-    fn from(mls_credential_type: MlsCredentialType) -> Self {
-        Credential {
-            credential_type: match mls_credential_type {
-                MlsCredentialType::Basic(_) => CredentialType::Basic,
-                MlsCredentialType::X509(_) => CredentialType::X509,
-            },
-            credential: mls_credential_type,
-        }
+    /// Get this serialized credential content.
+    ///
+    /// This is the content of the `select` statement. It is a TLS serialized
+    /// vector.
+    pub fn serialized_content(&self) -> &[u8] {
```

Comment:
It would be nice if we could make this a bit more ergonomic. Would it be possible to make this function generic in the type of credential (which has to support `tls_codec::Deserialize`) and then try to deserialize internally? That would return a Result, but it would spare the caller the additional explicit deserialization step. Does that make sense?

Reply:
I added a second function below to get it deserialized. Is that what you mean?
I agree, that's a nice one to have. But I'd keep this too.

Reply:
Yes, that's what I meant, thanks!